File: pubnub-java/src/main/java/com/pubnub/internal/endpoints/DeleteMessagesImpl.java
package com.pubnub.internal.endpoints;

import com.pubnub.api.endpoints.DeleteMessages;
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
import com.pubnub.api.models.consumer.history.PNDeleteMessagesResult;
import com.pubnub.internal.InternalPubNubClient;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@Setter
@Accessors(chain = true, fluent = true)
public class DeleteMessagesImpl extends DelegatingEndpoint<PNDeleteMessagesResult> implements DeleteMessages {
    private List<String> channels;
    private Long start;
    private Long end;

    public DeleteMessagesImpl(InternalPubNubClient pubnub) {
        super(pubnub);
        channels = new ArrayList<>();
    }

    @Override
    protected ExtendedRemoteAction<PNDeleteMessagesResult> createAction() {
        return pubnub.deleteMessages(
                channels,
                start,
                end
        );
    }
}

File: pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/endpoints/DeleteMessagesImpl.kt
package com.pubnub.internal.endpoints

import com.pubnub.api.endpoints.DeleteMessages
import com.pubnub.api.models.consumer.history.PNDeleteMessagesResult
import com.pubnub.internal.EndpointImpl
import com.pubnub.internal.PubNubImpl

/**
 * @see [PubNubImpl.deleteMessages]
 */
class DeleteMessagesImpl internal constructor(deleteMessages: DeleteMessagesInterface) :
    DeleteMessagesInterface by deleteMessages,
    DeleteMessages,
    EndpointImpl<PNDeleteMessagesResult>(deleteMessages.configuration)